### PR TITLE
fix(filterFilter): Fix filtering using an object expression when the filter value is undefined

### PR DESCRIPTION
**Pull request by caseyhoward**
Fixes #10419

### CODE REVIEW

File: src/ng/filter/filter.js [hunk 1]
```diff
@@ -196,7 +196,7 @@ function deepCompare(actual, expected, comparator, matchAgainstAnyProp) {
       } else if (expectedType === 'object') {
         for (key in expected) {
           var expectedVal = expected[key];
-          if (isFunction(expectedVal)) {
+          if (isFunction(expectedVal) || isUndefined(expectedVal)) {
```
Comment (zuzusik) [inline src/ng/filter/filter.js]: This broke things in our project. It changed the behaviour of `$filter`. I don't think it is OK to change the behaviour in minor releases.

Comment (pkozlowski-opensource) [inline src/ng/filter/filter.js]: @zuzusik I was under the impression that c62fa6bd898e1048d4690d41034489dc60ba6ac2 was only merged to 1.4.x beta so a release where we've got breaking changes. Do you see it on any of the 1.3.x branches as well?

Comment (zuzusik) [inline src/ng/filter/filter.js]: It was successfully landed to `1.3.15`, look at the [changelog](https://github.com/angular/angular.js/blob/master/CHANGELOG.md#1315-locality-filtration-2015-03-17)

Comment (zuzusik) [inline src/ng/filter/filter.js]: @pkozlowski-opensource here it is: https://github.com/angular/angular.js/commit/63b9956faf4c3679c88a9401b8ccbb111c0294ee

**Comment by petebacondarwin** [comment]
This change was actually fixing a **regression** that occurred between 1.3.5 and 1.3.6 (see https://github.com/angular/angular.js/issues/10419). I am sorry that it took so long for us to fix this regression that you started using the filter in its regressed state.

File: src/ng/filter/filter.js (hunk 1 above)
Comment (petebacondarwin) [inline src/ng/filter/filter.js]: @zuzusik - This change was actually fixing a regression that occurred between 1.3.5 and 1.3.6 (see #10419). I am sorry that it took so long for us to fix this regression that you started using the filter in its regressed state.

**Pull request closed without merging**